src/rpc/connection/crypto.c

```diff
@@ -446,6 +454,16 @@ int crypto_recv_initiate(struct crypto_context *cc, unsigned char *data)
 
   cc->state = TUNNEL_ESTABLISHED;
 
+  char *pluginkey_base16_encoded = MALLOC_ARRAY((PLUGINKEY_SIZE*2) + 1, char);
```

Comment:
> why use heap memory? 

Stack is completely fine. 

> and why (PLUGINKEY_SIZE*2) + 1?

PLUGINKEY_SIZE represents the 8 bytes of the pluginkey. The string representation consists of 16 characters plus the \0 byte.